configuration.go

```diff
@@ -0,0 +1,72 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ecdysis
+
+import (
+	"reflect"
+
+	"github.com/spf13/viper"
+)
+
+type Config struct {
+	EnvPrefix  string
+	ParsedCfg  any
+	DefaultCfg any
```

Comment:
ParsedCfg and DefaultCfg should be of the same type, right? If so, we should try to enforce that, if it's not too complicated. Either through generics or another check.

Reply:
https://github.com/ConduitIO/ecdysis/pull/3/commits/4808a02dd48126a8b6521b5c6aada9ea3c34c9f2